src/nvim/message.c

```diff
@@ -3049,9 +3069,11 @@ int msg_end(void)
    * we have to redraw the window.
    * Do not do this if we are abandoning the file or editing the command line.
    */
-  if (!exiting && need_wait_return && !(State & CMDLINE)) {
-    wait_return(FALSE);
-    return FALSE;
+  if (msg_enable_ext()) {
```

Comment:
The failure from https://github.com/neovim/neovim/pull/16480#issuecomment-988403007 might be related to this condition. It probably makes sense to change this `msg_enable_ext` to `msg_enable_msgfunc` so that the old behavior is preserved for UI.

Reply:
Oh, it breaks current behavior.